rust/core-lib/src/editor.rs

```diff
@@ -904,6 +886,8 @@ impl Editor {
         let offset = self.view.line_col_to_offset(&self.text, line as usize, col as usize);
         match ty {
             GestureType::ToggleSel => self.view.toggle_sel(&self.text, offset),
+            GestureType::MultiLineSelect => self.view.select_line(&self.text, offset, line as usize, true),
+            GestureType::MultiWordSelect => self.view.select_word(&self.text, offset, true)
```

Comment:
Would it be more consistent to name these `GestureType`s to:
```
SelectLine
SelectWord
```?

And here comes the newbie question:
Why is this necessary at all?

Reply:
Regarding the why: I added support for multi selection of lines and words in xi-mac (see this pull-request: https://github.com/google/xi-mac/pull/159). So every time the gestures are executed in xi-mac (either by pressing <kbd>CMD</kbd> and clicking twice or three times) then xi-mac sends the RPCs to the core. The RPC messages contain as gesture type `multi_line_select` or `multi_word_select`. For message handling xi-editor will eventually deserialize the messages from JSON and `multi_line_select` will become `MultiLineSelect` and `multi_word_select` will become `MultiWordSelect`. Depending on the gesture type, which is determined at this specific point in the code, the corresponding actions (`self.view.select_line` or `self.view.select_word`) are performed. 

Regarding the name, these events are triggered when users press <kbd>CMD</kbd> + double click or <kbd>CMD</kbd> + triple click (on OSX <kbd>CMD</kbd>, on other platforms maybe <kbd>CTRL</kbd>). So I thought when users perform these gestures their intent is specifically to select multiple words/lines. 
`self.view.select_word` and `self.view.select_line` are also used for selecting single lines/words in `do_click`. The last parameter of these functions indicates whether multiple or single lines/words can be selected.

Reply:
@scholtzan thanks for the detailed explanation. Now it makes perfectly sense. I'll add the corresponding gestures to xi-gtk. :)